Add v5 upgrade guide (#160)
humanmade/altis-documentation: other-docs/guides/upgrading/v5.md

If you use Local Chassis you will need to also update Chassis and its extensions by running `composer chassis upgrade`.

## Breaking Changes

### SEO

The AMP and Facebook Instant Articles plugins have been unbundled from Altis. This allows you to specify the versions of each directly in your Composer requirements. While these are no longer included with Altis, we plan to maintain compatibility with AMP in new features as necessary.

If you're using AMP in your project, after updating your Altis requirements, you will need to additionally require the AMP plugin:

```sh
$ composer require ampproject/amp-wp@1
```

Note that while Altis v4 bundles version 1 of the AMP plugin, version 2 is now available with [many major changes](https://github.com/ampproject/amp-wp/releases/tag/2.0.0), so consider updating at the same time.

If you're using Facebook Instant Articles, you will need to require the FBIA plugin. This requires first [adding WordPress Packagist as a repository](docs://getting-started/third-party-plugins#managing-plugins-via-composer), then requiring the plugin from wpackagist:

```sh
$ composer require wpackagist-plugin/fb-instant-articles
```

> 

Reviewer: If we add core sitemaps in that can go here too

Author: Can handle in follow-up once merged?

Fuzzy search can now be [configured](docs://search/search-configuration.md#fuzzy-matching). This allows tuning the allowed edit distance, prefix length, expansions and transpositions at a granular level to match desired behaviour.

Other changes included in the update to ElasticPress 3.4 also include the ability for users to adjust individual search queries, allowing results to be manually tuned or overridden, as well as better WooCommerce integration out of the box.

Reindexing is not required for these changes. When user dictionaries are changed, any relevant content will be automatically reindexed.

### Suggestions when searching

Search suggestions can now be retrieved from Elasticsearch using autosuggest mode. Autosuggest mode matches partial words rather than whole words, which improves relevance while typing.

Autosuggest mode also matches against a subset of post fields rather than all fields, and defaults to true when the current request is an `admin-ajax.php` request (i.e. `DOING_AJAX === true`).

Consult [the query integration documentation](docs://search/cms-query-integration.md) for further details on how to use and control this functionality.

> This feature does not require reindexing, however due to changes in the stemmer, relevance of autosuggest results can be improved by reindexing. This can be [performed at your discretion](docs://search/reindexing-content.md).

Author: Did the default config for that change?

> This feature does not require reindexing, however due to changes in the stemmer, relevance of autosuggest results can be improved by reindexing. This can be [performed at your discretion](docs://search/reindexing-content.md).

Reviewer: No sorry I'm getting my wires crossed - they're actually separate. The `autosuggest` in config is really just for adding the client side search form dropdown now.

This does require reindexing though as it needs an updated mapping.

> This feature does not require reindexing, however due to changes in the stemmer, relevance of autosuggest results can be improved by reindexing. [Reindexing can be performed at your discretion](docs://search/reindexing-content.md).

Reviewer: ```suggestion
This feature does require reindexing as it makes changes to the field mapping. [Reindexing can be performed at your discretion](docs://search/reindexing-content.md).
```

Additionally, data reported to X-Ray has been improved to remove noise and provide more accurate data. Internal healthcheck requests sent by Altis Cloud infrastructure are no longer reported in X-Ray, and database queries now use the final query sent to the database.

## Other Features and APIs

### Post Cloning

The CMS module now includes the ability to clone posts, including their full post data and post metadata. This behaviour can be modified via the [provided actions and filters](docs://cms/post-cloner.md).

### Post GUIDs

In order to set the groundwork for future import/export and migration tooling, GUIDs for post objects in Altis now use UUID URNs (e.g. `urn:uuid:d72fdc38-1305-4568-9821-463120b250f3`). This replaces the former URL-style GUIDs used for posts (e.g. `http://altis.local/?p=42`).

Due to legacy use of the GUID field, some older code may attempt to use the GUID for image media source URLs. This code must be updated to use `wp_get_attachment_url()` instead, which will also ensure that Tachyon and the Altis CDN are correctly used.

New code should only use the GUID field to uniquely identify posts, and never as a URL.

### Minimum version of the AWS SDK

> The minimum version of the AWS SDK included with Altis has been bumped to 3.150.0. We recommend keeping the SDK up-to-date via Dependabot.

Reviewer 2: So we recommend this? Seems like a pain to call out specifically.

Author: Might drop this section entirely as it seems to be backwards compatible, but had included as an "FYI our minimum of this dep you might also be using is being bumped"

### Multisite

The sites list in the Network Admin now displays a more user-friendly view of sites, using site names rather than domains. Additionally, some irrelevant options (such as the mark-as-spam ability) are now de-emphasised in the UI.

### Development Environments

Compatibility issues with using XDebug in Local Server on a Linux host have been fixed. The remote host configuration option is now set automatically for you internally in order to correctly route connections. Consult the [XDebug documentation](docs://local-server/using-xdebug.md) for more information on how to use it in your editor.

### Other Changes

> Altis v5 includes many other small changes, consult the changelog for full details of all the changes shipped in this release.

Reviewer: GUessing we link those changelogs up once the stable is released?
